daml-assistant/daml-helper/src/DA/Daml/Helper/Start.hs

```diff
@@ -228,26 +228,29 @@ runStart
     scriptOpts <- withOptsFromProjectConfig "script-options" scriptOpts projectConfig
     doBuild
     doCodegen projectConfig
-    listenForKeyPress projectConfig
     let scenarioArgs = maybe [] (\scenario -> ["--scenario", scenario]) mbScenario
     withSandbox sandboxClassic sandboxPort (darPath : scenarioArgs ++ sandboxOpts) $ \sandboxPh sandboxPort -> do
+        let doRunInitScript =
+              whenJust mbInitScript $ \initScript -> do
+                  putStrLn "Running the initialization script."
+                  procScript <- toAssistantCommand $
+                      [ "script"
+                      , "--dar"
+                      , darPath
+                      , "--script-name"
+                      , initScript
+                      , if any (`elem` ["-s", "--static-time"]) sandboxOpts
+                          then "--static-time"
+                          else "--wall-clock-time"
+                      , "--ledger-host"
+                      , "localhost"
+                      , "--ledger-port"
+                      , case sandboxPort of SandboxPort port -> show port
+                      ] ++ scriptOpts
+                  runProcess_ procScript
+        doRunInitScript
```

Comment:
We now run the init script before navigator is up. I guess there is no reason not to do this?

Reply:
I couldn't think of any. In fact, I think it's safer to make sure the navigator reflects the most up to date state.